### ISSUE TITLE
transform error with validation data.

### ISSUE THREAD

**Issue by DayanSiddiquiNXD** [awslabs/gluon-ts#1187]
using gluonts version 0.6.0 on linux. when i try to use a validation dataset with my model, it throws this error

> Exception: Reached maximum number of idle transformation calls.
>This means the transformation looped over GLUONTS_MAX_IDLE_TRANSFORMS=100 inputs without returning any output.
>This occurred in the following transformation:
>gluonts.transform.split.InstanceSplitter(dummy_value=0.0, forecast_start_field="forecast_start", future_length=9, is_pad_field="is_pad", lead_time=0, output_NTC=True, past_length=166, pick_incomplete=True, start_field="start", target_field="target", time_series_fields=["time_feat", "observed_values"], train_sampler=gluonts.transform.sampler.ExpectedNumInstanceSampler(num_instances=1.0))

the validation and train data are being split across time series over the time dimension (i.e. the last n observations for each time series are the val, the ones before are train)

this is the specific code which loops over the data for each time series:
`    val_df = full_group[(full_group['timeIndex'] > train_until) & (full_group['timeIndex'] <= test_until)]
    val_dicts.append({'start': val_start, 'target': val_df['Volume']})`

where train_until is also the end point for the training data, train_until < test_until so val_df is not empty, and the pred is defined as:
`prediction_length = test_until - train_until`

and this constructs the dataloader afterwards:
`val_data = ListDataset(
    val_dicts,
    freq = freq
    )`

after which the model is trained
`estimator_15 = DeepAREstimator(freq=freq, prediction_length=prediction_length, trainer=Trainer(epochs=15))
predictor_15 = estimator_15.train(training_data=training_data, validation_data = val_data)`

which is where the error occurs, at the line with the predictor

THE REASON I'M BRINGING THIS HERE is that when I apply an overlap in the val and train data, if seems to fix the issue. which is strange unless i'm missing something about prediction_length.

`overlap = 1`
`val_df = full_group[(full_group['timeIndex'] > train_until - overlap) & (full_group['timeIndex'] <= test_until)]`

and the error is fixed in a probabilistic manner. in that an overlap of 1 almost always still results in the error, while an overlap of 13 observations almost always avoids the error. just from an eye test it seems like overlaps closer to 1 still have a higher chance of throwing this bug, which drops off with a larger overlap.

i just wanna monitor val data loss to prevent overfitting, which is not really possible if the model is training on a section of the val dataset

**Comment by jgasthaus** [comment]
This is not a bug, but the expected and intended behavior.

You need to use 
```
val_df = full_group[(full_group['timeIndex'] <= test_until)]
```

i.e. include the *full* training time range validation data set. This will do exactly what you want, namely evaluate only on the last ```prediction_length``` time steps of the validation data set.

The reason for this is that in GluonTS the train/validation/test data sets do not have to contain the same time series, so GluonTS does not assume that the i-th series in the validation set corresponds to the i-th series in the your training set. Because of this, you need to provide the full series in the validation set, so it can use the data before the last ```prediction_length``` points as context for making the prediction.

**Comment by DayanSiddiquiNXD** [comment]
@jgasthaus thank alot. this makes sense. i guess i was right when i said "which is strange unless i'm missing something about prediction_length". my understanding about this was just incomplete. this is successfully closed
